File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
from collections import defaultdict
import heapq
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # pairs of values and frequency in the priority queue
    
        m = defaultdict(int)
        for i in nums: 
            m[i]+=1
        
        vec = [(-v,k)  for (k,v) in m.items()]
        heapq.heapify(vec)
        f = []
        for i in range(k):
            f.append(heapq.heappop(vec)[1])
        return f

File: Data Structures & Algorithms/maximum-subarray/submission-4.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max = nums[0];
        int count = max;
        for(int i =1; i < nums.size(); i++){
            
            if(count < 0 && nums[i] > count){
                count = nums[i];
            }else {count += nums[i];}

            if(count > max){
                max = count;
            }
            if(count < 0 && count < max){
                count = 0;
            }
            
        }
        return max;

    }
};
